clients/src/main/java/org/apache/kafka/clients/admin/Config.java

```diff
@@ -36,14 +37,14 @@
      * Create a configuration instance with the provided entries.
      */
     public Config(Collection<ConfigEntry> entries) {
-        this.entries = Collections.unmodifiableCollection(entries);
+        this.entries = Objects.requireNonNull(entries);
```

Comment:
I wonder if we should copy to a list so that we are guaranteed that the collection won't change behind our backs and so that comparison between two `Config` objects will be valid even if they are constructed using different collection types.

Reply:
On second thought, considering the usage of `get` below, maybe we should create a `Map` instead so that we can more easily access an entry by name.

Reply:
I thought about a defensive copy - that would be my standard pattern here. But wanted to minimise my changes.

Give me a mo... I'll switch...